### PR TITLE
fix(api): manage variable deletion and env keys as code

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18862.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: engine/api/environment/environment_importer.go [hunk 1]
```diff
@@ -67,45 +67,25 @@ func Import(db gorpmapper.SqlExecutorWithTx, proj sdk.Project, env *sdk.Environm
...
-		for j := range into.Variables {
-			log.Debug(ctx, "ImportInto> \t with >> %s", into.Variables[j].Name)
-			if env.Variables[i].Name == into.Variables[j].Name {
-				env.Variables[i].ID = into.Variables[j].ID
-				found = true
-				updateVar(&env.Variables[i])
-				break
-			}
+		if err := InsertVariable(db, into.ID, &env.Variables[i], u); err != nil {
+			return err
```
Comment (fsamin) [inline engine/api/environment/environment_importer.go:82]: no more msgChan <- sdk.NewMessage ?

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18863.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18864.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: engine/api/environment/environment_importer.go (hunk 1 above)
Comment (sguiheux) [reply engine/api/environment/environment_importer.go:83]: fixed :)

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18870.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Review by fsamin — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

**Pull request merged**